### PR TITLE
miq_ae_engine: deliver method refactoring.

### PR DESCRIPTION
**Pull request by pkomanek**
* Deliver method refactoring
* Adding the missing specs for miq_ae_engine 
* Removing the legacy code.

### CODE REVIEW

**Comment by gmcculloug** [comment]
cc @tinaafitz

**Comment by mkanoor** [comment]
@pkomanek 
can you run it with 

```
export CI=1
rspec ...
````
and see if every line of code is getting tested for this file.
I am not sure if we have many tests around this code.

**Comment by mkanoor** [comment]
@pkomanek 
Is every line of the code getting executed, can you post the output of

export CI=1
rspec ...

There is an html file that will get generated, I just want to ensure that all code paths have been executed.

**Comment by pkomanek** [comment]
@mkanoor 
Code coverage for miq_ae_enginge.rb is 98.39%. Lines 155, 168 and 214 are not tested yet. Should I make it tested?

**Comment by mkanoor** [comment]
@pkomanek 
Can we add specs for those, I think 2 of them relate to MiqRequest and one is a simple data type.

File: lib/miq_automation_engine/engine/miq_ae_engine.rb [hunk 1]
```diff
@@ -27,71 +27,63 @@ def self.deliver_queue(args, options = {})
...
+      object_name = "#{options[:object_type]}.#{options[:object_id]}"
       _log.info("Delivering #{options[:attrs].inspect} for object [#{object_name}] with state [#{state}] to Automate")
-      automate_attrs = options[:attrs].dup
+      automate_attrs = automate_attrs_from_options(options)
 
-      if object_type
-        vmdb_object = object_type.constantize.find_by(:id => object_id)
-        automate_attrs[create_automation_attribute_key(vmdb_object)] = object_id
+      if options[:object_type]
+        vmdb_object = options[:object_type].constantize.find_by(:id => options[:object_id])
```
Comment (gmcculloug) [inline lib/miq_automation_engine/engine/miq_ae_engine.rb]: @pkomanek Part of this refactoring was to incorporate the change to use `find_by!` from https://github.com/ManageIQ/manageiq-automation_engine/pull/84/files on this line and make it easier to test.

Can you create a follow-up PR to introduce this change and tests?

**Pull request merged**